bundle/phases/initialize.go

```diff
@@ -107,6 +109,7 @@ func Initialize(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
 
 		apps.Validate(),
 
+		mutator.ValidateTargetMode(),
```

Comment:
This inverts the ordering.

Before, validation happened before transformation/applying defaults. Now it happens after. Can you confirm there are no adverse side effects that can cause errors where first there weren't any? @lennartkats-db @kanterov

Reply:
We have found it more straightforward to move all validation into ValidateTargetMode instead of doing it in 2 steps.